addons/mail/controllers/mail.py

```diff
@@ -236,6 +239,14 @@ def mail_thread_message_redirect(self, message_id, **kwargs):
                 return request.redirect(f'/web/login?redirect=/mail/message/{message_id}')
             raise Unauthorized()
 
+        if (
+            (data := self._get_public_chatter_url_data(message)) and
+            message.message_type == "comment" and
+            not request.env.user._is_internal()
+        ):
+            url_params = {**data.get("url_params", {}), "highlight_message_id": message_id}
+            return request.redirect(f"{data['url']}?{url_encode(url_params)}")
```

Comment:
I don't remember the state of the discussion but can't we just always redirect to `mail_action_view` instead of adding new code here?